tests/queries/0_stateless/02907_backup_mv_with_no_source_table.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+CUR_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CUR_DIR"/../shell_config.sh
+
+${CLICKHOUSE_CLIENT} -nm --query "
+drop table if exists src;
+create table src (a Int32) engine = MergeTree() order by tuple();
+
+drop table if exists dst;
+create table dst (a Int32) engine = MergeTree() order by tuple();
+
+drop table if exists mv;
+create materialized view mv to dst (a Int32) as select * from src;
+"
+
+${CLICKHOUSE_CLIENT} -nm --query "
+drop table src;
```

Comment:
What if you also drop `dst` here?

Reply:
Then there will be no error because there is a check with `create.to_table_id`:
https://github.com/ClickHouse/ClickHouse/blob/eb0e4a0aa5d38f868c134fbf4f0d94335f6684c4/src/Interpreters/InterpreterCreateQuery.cpp#L1183-L1189
Anyway here I just made behavior consistent with table's attach - we also do not perform this check if `create.attach`, so a user can drop source table and table will survive server restart without any error, so makes sense not to perform this check on backup restore as well.

Reply:
Let's drop both `src` and `dst` in the test then?

Reply:
But if we drop `dst` then without the fix the test will pass, so it means the test does not test.

Reply:
ah, or you mean adding one more test case, e.g. having a test when only src is dropped + another test when both src and dst are dropped? this will be better

Reply:
I see. There are two different test cases. Ok then.